apps/explorer/lib/explorer/exchange_rates/source.ex

```diff
@@ -56,17 +57,63 @@ defmodule Explorer.ExchangeRates.Source do
     end
   end
 
+  defp remove_market_cap_if_certain_token(value, address) do
+    if address in ["0x49d5c2bdffac6ce2bfdb6640f4f80f226bc10bab", "0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7"] do
+      Map.delete(value, "usd_market_cap")
+    else
+      value
+    end
+  end
+
+  defp swap_address(result, original_address, new_address) do
+    case Map.has_key?(result, original_address) do
+      true ->
+        {value, remaining_result} = Map.pop(result, original_address)
+        value_updated = remove_market_cap_if_certain_token(value, original_address)
+        Map.put(remaining_result, new_address, value_updated)
+
+      false ->
+        result
+    end
+  end
+
+  @spec get_exchange_rate(String.t()) :: Token.t() | nil
+  defp get_exchange_rate(symbol) do
+    ExchangeRates.lookup(symbol)
+  end
+
+  defp update_result_addresses(result) do
+
+    result
+    |> swap_address("0x49d5c2bdffac6ce2bfdb6640f4f80f226bc10bab", "0x2c2E0B0c643aB9ad03adBe9140627A645E99E054") # weth
+    |> swap_address("0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7", "0x6318374DFb468113E06d3463ec5Ed0B6Ae0F0982") # wrapped-avax
+    |> swap_address("0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e", "0xCc44eB064CD32AAfEEb2ebb2a47bE0B882383b53") # usd-coin
+    |> swap_address("0x9702230a8ea53601f5cd2dc00fdbc13d4df4a8c7", "0xA167bcAb6791304EDa9B636C8beEC75b3D2829E6") # tether
+    |> swap_address("0xd586e7f844cea2f87f50152665bcbc2c279d8d70", "0x38C2a6953F86a7453622B1E7103b738239728754") # dai
+    |> swap_address("0x5947bb275c521040051d82396192181b413227a3", "0xDF8DBA35962Aa0fAD7ade0Df07501c54Ec7c4A89") # chainlink
+    |> swap_address("0x50b7545627a5162f82a992c33b87adc75187b218", "0x1d7fb99AED3C365B4DEf061B7978CE5055Dfc1e7") # wrapped-bitcoin
```

Comment:
It would be useful to have a single reference to these addresses to know what they are. For example, some of these addresses are also used in the `remove_market_cap_if_certain_token` method.

Maybe something like https://hexdocs.pm/elixir/1.16/module-attributes.html#as-constants

Reply:
I don't want to overcomplicate it and I'm not sure this is possible, but is there a way to use env variables for the addresses to replace?
The benefits would be that it can work in both testnet and mainnet, and if a new token were to be added, it wouldn't require a code change.

Reply:
Agree, I refactored the previous implementation to use environment variables instead of hardcoded values. 
The current implementation use the following variables:
- `HORIZEN_EON_TOKEN_LIST_TO_FETCH` - comma separated list of pairs _<token-name _address>_ (ex. `weth_0x2c2E0B0c643aB9ad03adBe9140627A645E99E054`)
- `TOKEN_ADDRESS_PAIRS_EXT_PLATFORM_EON` - comma separated list of pairs _<external-platform-address_eon_address>_ (ex. `0x49d5c2bdffac6ce2bfdb6640f4f80f226bc10bab_0x2c2E0B0c643aB9ad03adBe9140627A645E99E054`)
- `EXT_PLATFORM_FOR_TOKEN_FETCH` - external platform used to retrieve token exchange rate from coingecko
- `WRAPPED_ZEN_ADDRESS` - wrapped zen address if the token is present in the sidechain

The general logic to retrieve the token exchange rate remain the same as before:
- call to coingecko `api/v3/coins/list` api to retrieve the list of tokens anf related platfoms
- adding of horizen-eon addresses to the previuos response
- call to coingecko `api/v3/simple/token_price/<external-platform>` to retrieve exchange rate on the external platform
- swap of the response addresses with the one present in the eon sidechain
- update of the token price 

Also added comments to the methods introduced with this PR.